packages/web/src/components/range/RangeInput.js

```diff
@@ -64,29 +58,34 @@ class RangeInput extends Component {
 			}
 		}
 
-		const currentValue = {
-			start: this.state.start,
-			end: this.state.end,
-			[name]: value,
-		};
-
-		const { value: valueProp, onChange } = this.props;
-		if (valueProp === undefined) {
+		if (this.isControlled) {
+			this.props.onChange({
+				...this.value,
+				[name]: value,
+			});
+		} else {
 			this.setState({
 				[name]: value,
 			});
-		} else if (onChange) {
-			onChange(currentValue);
 		}
 	};
 
-	handleSliderChange = ([start, end]) => {
-		const { value, onChange } = this.props;
+	get isControlled() {
+		return this.props.onChange;
+	}
+
+	get value() {
+		return {
+			start: this.isControlled ? Number(this.props.value.start) : Number(this.state.start),
```

Comment:
Any specific reason that we are explicitly converting type for `this.props.value.start` & `this.props.value.end` ?

`value` prop type would be validated by `range` type. Check [here](https://github.com/appbaseio/reactivecore/blob/master/src/utils/types.js#L60)

Reply:
I think I was preserving existing behavior: https://github.com/appbaseio/reactivesearch/pull/897/commits/1cdce7188605f6cb4ee3086172b7ab56d5c8c610#diff-0b89c55cd09ffb0b7b8c09e1bef3dabfL120

But this is a really good question. Is it a better or worse API if the component does this type coercion automatically? 

On the one hand, I find it nice as a user to have the component work despite the type warnings (provided the type can be coerced to a number). 

But on the other hand, maybe this coercion could cause weird `NaN` failures if the user entered a string that can't be coerced? Granted the component probably wouldn't have behaved well anyway, and I generally prefer to uncover errors early.

My instinct is to probably lean on the side of coercing simply because I mostly see it helping over hurting. I'm curious to hear your thoughts

Reply:
So I thought of one interesting and sort of annoying consequence of this type coercion. We can't completely blank out an input field using the delete key. The input will automatically be coerced to 0 if we ever try to blank it out.

Reply:
I think that is how it should be. We want to make sure that there is some validation for numerical inputs and hence I agree with you on this - it's completely fine to have coercion in place. It's definitely helping over hurting here.

Reply:
I didn't understand/represent the problem well before. I think we have two scenarios to talk about:

1. The value that is passed to the `RangeSlider`
2. The value passed to the two `input` html elements

I think it makes a lot of sense to coerce the value passed to `RangeSlider` since it will certainly reduce error. 

I'm not so sure we should coerce the value to the `input` elements, primarily because it will prevent users from emptying the inputted value. No matter how many times the user clicks backspace, the value will end up at 0, never `null` or `""`. This creates an awkward experience when trying to type in a custom value. For example, if you delete the existing input and then type `10`, the entered value will actually be `010`. This is definitely not the behavior I would expect as a user.

If there are no objections, I'm going to coerce the value passed to the `RangeSlider` but not the value passed to the input elements.

Reply:
That's a valid point. You're right - maybe we should coerce the value passed to the RangeSlider and not the value passed to the input elements. 

Maybe we should add a input field validation here to take only numbers or empty values ~ so that we have a visual feedback of what the component expects from the user. What do you think?

Reply:
yeah, I think that's a good idea. Should we do that in this PR?